contracts/UpkeepRegistrationRequests.sol

```diff
@@ -151,6 +162,38 @@ contract UpkeepRegistrationRequests is Owned {
         );
     }
 
+    /**
+     * @notice deny will delete a registration request hash and refund the user
+     * @param name the name up the upkeep
+     * @param upkeepContract the upkeep target address
+     * @param gasLimit the upkeep's execute gas limit
+     * @param adminAddress the admin of the upkeep
+     * @param checkData the data to pass to the checkUpkeep function
+     * @param amount the amount of LINK to initially fund the upkeep with
+     * @param refundAddress the address of the user to refund LINK to in case the request is denied
+     * @param hash the request hash
+     */
+    function deny(
+        string memory name,
+        address upkeepContract,
+        uint32 gasLimit,
+        address adminAddress,
+        address refundAddress,
+        bytes calldata checkData,
+        uint96 amount,
+        bytes32 hash
+    )
```

Comment:
alternatively, we could keep the refundAddress and amount in a mapping in storage, rather than re-compute and verify the hash here. That would simplify this function signature to deny(bytes32 hash). But I thought this would be cheaper to just re-generate the hash here.